pkg/provisioner/controller.go

```diff
@@ -638,3 +655,43 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+	if reflect.DeepEqual(ob.Spec.Endpoint.AdditionalConfigData, obc.Spec.AdditionalConfig) {
+		return nil
+	}
+	log.Info("Additional config changed, hence updating OB")
+	tmp := make(map[string]string)
+
+	overwriteAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData, tmp)
+	overwriteAdditionalConfig(obc.Spec.AdditionalConfig, ob.Spec.Endpoint.AdditionalConfigData)
+
+	err = c.provisioner.Update(ob)
+	if err != nil {
+		log.Error(err, "updating from Provisioner failed")
+		return err
+	}
+	ob, err = updateObjectBucket(ob, c.libClientset)
+	if err != nil {
```

Comment:
It seems like this is a good place to do a retry inline vs. requeuing on error. It's potentially expensive for the provisioner to update the new additionalConfig, then update back to the old value, then the event is requeued, then provisioner updates to new config again, etc.

Reply:
are suggesting the following
`for i=0; i<3 &&  err !=nil; i++ {
ob, err = updateObjectBucket(ob, c.libClientset)
}

Reply:
No. Look at examples of using the apimachinary's _wait_ package; (PollImmediate)[https://github.com/kubernetes/apimachinery/blob/v0.21.0/pkg/util/wait/wait.go#L440]. Eg.
```
   err = wait.PollImmediate(2*time.seconds, 2*time.minutes, func() (done bool, err error) {
             _, err = updateObjectBucket(ob, c.libClientset)
    	     if err != nil {
		     log.Error(err, "updating OB failed, retrying...")
		     return false, err
	     }
             return true, nil
    })

    if err != nil {
    	log.Error(err, "updating OB failed")
	// revert to old values
	overwriteAdditionalConfig(tmp, ob.Spec.Endpoint.AdditionalConfigData)
	err = c.provisioner.Update(ob)
    }
    return err
}
```